lib/devise/multi_email/models/authenticatable.rb

```diff
@@ -69,27 +50,18 @@ def find_first_by_auth_conditions(tainted_conditions, opts = {})
 
           if email && email.is_a?(String)
             conditions = filtered_conditions.to_h.merge(opts).
-              reverse_merge(emails: { email: email })
+              reverse_merge(_multi_email_reflect_on_emails_association.table_name => { email: email })
 
-            resource = joins(:emails).find_by(conditions)
+            resource = joins(_multi_email_emails_association_name).find_by(conditions)
             resource.current_login_email = email if resource.respond_to?(:current_login_email=)
             resource
           else
             super(tainted_conditions, opts)
           end
         end
 
-        def email_class
-          email_association = reflect_on_association(EMAILS_ASSOCIATION)
-          if email_association
-            email_association.class_name.constantize
-          else
-            raise "#{self.class.name}: Association :#{EMAILS_ASSOCIATION} not found: It might because your declaration is after `devise :multi_email_confirmable`."
-          end
-        end
-        
         def find_by_email(email)
-          joins(:emails).where(emails: {email: email.downcase}).first
+          joins(_multi_email_emails_association_name).where(_multi_email_reflect_on_emails_association.table_name => {email: email.downcase}).first
```

Comment:
@allenwq the `joins` method takes the association name, but the `where` method needs to use the table name when using Hash conditions. Or am I looking at the wrong place?